viite-UI/src/utils/backend-utils.js

```diff
@@ -143,6 +143,26 @@
         });
     });
 
+    this.insertNewRoadLink = _.throttle(function(data, success, failure) {
+      var Json = {
+        linkIds : data[0],
```

Comment:
I might be wrong but isn't it possible to use data.linkid instead of position based matching?